### PR TITLE
additional mainnet difficulty tests fix

### PR DESCRIPTION
**Pull request by winsvega**

### CODE REVIEW

File: test/unittests/libethcore/difficulty.cpp [hunk 1]
```diff
@@ -214,7 +214,7 @@ BOOST_AUTO_TEST_CASE(difficultyTestsCustomMainNetwork)
 
 	if (dev::test::Options::get().filltests)
 	{
-		u256 homesteadBlockNumber = 3500000;
+		u256 homesteadBlockNumber = 4300000;
```
Comment (pirapira) [inline test/unittests/libethcore/difficulty.cpp]: Why?

Comment (winsvega) [reply test/unittests/libethcore/difficulty.cpp]: ok i haven't mention that. but this blocknumber is byzantine transition number

Comment (winsvega) [reply test/unittests/libethcore/difficulty.cpp]: it used to be the test cases around homestead number. now this is around byzantine

Comment (pirapira) [reply test/unittests/libethcore/difficulty.cpp]: Please change the name of the variable.

File: test/unittests/libethcore/difficulty.cpp [hunk 2]
```diff
@@ -42,14 +42,14 @@ std::string const c_testDifficulty = R"(
 		"currentDifficulty" : "[CDIFF]"
 	},
 )";
+h256 const someHash = sha3("whatever nonempty string");
```
Comment (pirapira) [inline test/unittests/libethcore/difficulty.cpp]: `nonzeroHash` is a better name, I think.

**Pull request merged**